xmppserver/src/main/java/org/jivesoftware/openfire/spi/RoutingTableImpl.java

```diff
@@ -1077,6 +1083,10 @@ public void leftCluster() {
             final NodeID nodeID = server.getNodeID();
```

Comment:
Recent changes made sure that the nodeId never changes. For performance reasons, you could wrap this whole block in `if(!nodeId.equals(defaultNodeID))`

Reply:
That's a great suggestion. I've added such a check.